aws/resource_aws_cloudwatch_dashboard.go

```diff
@@ -0,0 +1,123 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCloudWatchDashboard() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudWatchDashboardPut,
+		Read:   resourceAwsCloudWatchDashboardRead,
+		Update: resourceAwsCloudWatchDashboardPut,
+		Delete: resourceAwsCloudWatchDashboardDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		// Note that we specify both the `dashboard_body` and
+		// the `dashboard_name` as being required, even though
+		// according to the REST API documentation both are
+		// optional: http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutDashboard.html#API_PutDashboard_RequestParameters
+		Schema: map[string]*schema.Schema{
+			"dashboard_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dashboard_body": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateJsonString,
+				StateFunc: func(v interface{}) string {
+					json, _ := normalizeJsonString(v)
+					return json
+				},
+				DiffSuppressFunc: suppressEquivalentJsonDiffs,
+			},
+			"dashboard_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Forgot to say that `dashboard_body` & `dashboard_name` should also be set as `ForceNew`, as those are required!

Also, according to the [documentation](http://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutDashboard.html) the name should be validated: ` The maximum length is 255, and valid characters are A-Z, a-z, 0-9, "-", and "_".`